File: app/src/main/java/ru/ibakaidov/distypepro/structures/Statement.java
package ru.ibakaidov.distypepro.structures;

import java.util.HashMap;

public class Statement {
    public String categoryId;
    public long created;
    public String id;
    public String text;

    public Statement(String id, String categoryId, String text, long created){

        this.id = id;
        this.categoryId = categoryId;
        this.text = text;
        this.created = created;
    }


    public static Statement fromHashMap(HashMap hashMap){
        long created = hashMap.containsKey("created")? (long) hashMap.get("created"):0;
        String id = (String) hashMap.get("id");
        String categoryId = (String) hashMap.get("categoryId");
        String text = (String) hashMap.get("text");

        return new Statement(id,categoryId, text, created);
    }
}

File: shared/src/commonMain/kotlin/ru/ibakaidov/distypepro/structures/Statement.kt
package ru.ibakaidov.distypepro.structures

data class Statement(
    val id: String,
    val categoryId: String,
    val text: String,
    val created: Long
) {
    companion object {
        fun fromMap(map: Map<*, *>): Statement {
            val created = (map["created"] as? Number)?.toLong() ?: 0L
            val id = map["id"] as? String ?: ""
            val categoryId = map["categoryId"] as? String ?: ""
            val text = map["text"] as? String ?: ""
            return Statement(id, categoryId, text, created)
        }
    }
}
